sources/react-app/src/shared/hooks.js

```diff
@@ -69,6 +69,248 @@ export function useNavigation() {
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+                authorBio_o {
+                  item {
+                    key
+                    component {
+                      guid: objectId
+                      contentTypeId: content__type
+                      label: internal__name
+                      path: localId
+                      bio_t
+                      name_s
+                      profilePic_s
+                      linkButtonText_s
+                      linkButtonUrl_s
+                      showLinkButton_b
+                      facebookLink_s
+                      twitterLink_s
+                      instagramLink_s
+                      youTubeLink_s
+                    }
+                  }
+                }
+                categories_o {
+                  item {
+                    key
+                    value_smv
+                  }
+                }
+                tags_o {
+                  item {
+                    key
+                    value_smv
+                  }
+                }
+              }
+            }
+          }
+        `
+      }).then(({ data }) => {
+        (!destroyedRef.current) && update({ posts: parseDescriptor(data.page_post.items) });
+      });
+    }
+  }, [update, posts, postsLoading]);
+  return posts;
+}
+
+export function useTaxonomies() {
+  const [{ taxonomies, taxonomiesLoading }, update] = useGlobalContext();
+  const destroyedRef = useRef(false);
+  useEffect(() => {
+    return () => {
+      destroyedRef.current = true;
+    };
+  }, []);
+  useEffect(() => {
+    if (!taxonomies && !taxonomiesLoading) {
+      update({ taxonomiesLoading: true });
+      fetchQuery({
+        text: `
+          query Taxonomies {
+            component_taxonomy {
+              total
+              items {
+                guid: objectId
+                path: localId
+                contentTypeId: content__type
+                dateCreated: createdDate_dt
+                dateModified: lastModifiedDate_dt
+                label: internal__name
+                items {
+                  item {
+                    key
+                    value
+                  }
+                }
+              }
+            }
+          }
+        `
+      }).then(({ data }) => {
+        (!destroyedRef.current) && update({ taxonomies: parseDescriptor(data.component_taxonomy.items) });
+      });
+    }
+  }, [update, taxonomies, taxonomiesLoading]);
+  return taxonomies;
+}
+
+export function useCategories() {
+  const taxonomies = useTaxonomies();
+  return taxonomies?.filter(taxonomy => taxonomy.craftercms.path.includes('categories.xml'))[0].items.item;
+}
+
+export function useTags() {
+  const taxonomies = useTaxonomies();
+  return taxonomies?.filter(taxonomy => taxonomy.craftercms.path.includes('tags.xml'))[0].items.item;
+}
```

Comment:
Avoid multiple request to taxonomies. Make sure only one request flies out but all get updated when back.